tests/ci/run_fips_tests.sh

```diff
@@ -4,37 +4,23 @@
 
 source tests/ci/common_posix_setup.sh
 
+echo "Testing AWS-LC in FIPS Release mode."
+fips_build_and_test -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=1
+
+# The AL2 version of Clang does not have all of the required artifacts for address sanitizer, see P45594051
+if [[ ("${CC}" == *clang*) && ($(uname -a) == *Ubuntu*) ]]; then
+  echo "Building with Clang and testing AWS-LC in FIPS RelWithDebInfo mode with address sanitizer."
+  fips_build_and_test -DASAN=1 -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_SHARED_LIBS=1
+else
+  echo "Not building with Clang and testing AWS-LC in FIPS RelWithDebInfo mode."
+  fips_build_and_test -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_SHARED_LIBS=1
+fi
```

Comment:
Nit: in CodeBuild batch script, env variable `AWSLC_NO_ASM_FIPS` is used to enable/disable the ASAM. If `Ubuntu` is used, maybe delete `AWSLC_NO_ASM_FIPS`, which is no longer used. See previous code.
```
if [[  "${AWSLC_NO_ASM_FIPS}" == "1" ]]; then
  # This dimension corresponds to boringssl CI 'linux_fips_noasm_asan'.
  # https://logs.chromium.org/logs/boringssl/buildbucket/cr-buildbucket.appspot.com/8852496158370398336/+/steps/cmake/0/logs/execution_details/0
  echo "Testing AWS-LC in FIPS OPENSSL_NO_ASM Release mode."
  fips_build_and_test -DASAN=1 -DOPENSSL_NO_ASM=1 -DCMAKE_BUILD_TYPE=Release
  # TODO(shang): add ASAN build with RelWithDebInfo if final FIPS build uses the option.
fi
```